java/flight/flight-core/src/main/java/org/apache/arrow/flight/auth2/GeneratedBearerTokenAuthHandler.java

```diff
@@ -33,26 +36,29 @@
       .expireAfterAccess(2, TimeUnit.HOURS)
       .build();
 
-  @Override
-  protected String getIdentityForBearerToken(String bearerToken) {
-    return bearerToIdentityCache.getIfPresent(bearerToken);
+  public GeneratedBearerTokenAuthHandler(BasicCallHeaderAuthenticator basicAuthHandler) {
+    super(basicAuthHandler);
   }
 
   @Override
-  public boolean validateBearer(String bearerToken) {
-    return bearerToIdentityCache.getIfPresent(bearerToken) != null;
+  protected String validateBearer(String bearerToken) throws Exception {
+    final String peerIdentity = bearerToIdentityCache.getIfPresent(bearerToken);
+    if (peerIdentity == null) {
+      throw CallStatus.UNAUTHENTICATED.toRuntimeException();
+    }
+    return peerIdentity;
   }
 
-  String registerBearer(AuthResult handshakeResult) {
-    final String bearerToken = handshakeResult.getBearerToken()
-        .orElseGet(() -> {
-          final UUID uuid = UUID.randomUUID();
-          final ByteBuffer byteBuffer = ByteBuffer.wrap(new byte[16]);
-          byteBuffer.putLong(uuid.getMostSignificantBits());
-          byteBuffer.putLong(uuid.getLeastSignificantBits());
-          return Base64.getEncoder().encodeToString(byteBuffer.array());
-        });
-    bearerToIdentityCache.put(bearerToken, handshakeResult.getPeerIdentity());
-    return bearerToken;
+  @Override
+  protected void registerBearer(AuthResult authResult) {
+    String bearerToken = authResult.getHeaderMetadata().getValue();
```

Comment:
It seems like this is the only place where getHeaderMetadata() is used outside of the AuthResult's implementation. So really, this function (and corresponding interface) only exist for the purpose of servicing this method. Which we don't want, we are making the interface more complex so that we can handle one use case.

Instead, why don't we change this code to call authResult.appendToOutgoingHeaders() with a dummy CallHeaders object. Then just inspect the CallHeaders object to see if it has the bearer token header.

Note that your current code doesn't validate that getValue() is actually returning a bearer token.

Reply:
Done, updated the code to call appendToOutgoingHeaders with dummy headers and parse the headers as suggested.